sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1529,7 +1529,7 @@ pi_result piPlatformsGet(pi_uint32 NumEntries, pi_platform *Platforms,
     PrintPiTrace = true;
   }
 
-  if (ZeDebug & ZE_DEBUG_CALL_COUNT) {
+  if ((ZeDebug & ZE_DEBUG_CALL_COUNT) && !ZeCallCount) {
```

Comment:
please move this initialization to the lambda function of the global static `ZeCallCount`

Reply:
While it would make it thread-safe, there will be no proper delete for that if piTearDown is not called. I guess we need a mutex to have this initialization here and thread-safe. You can also probably use std::call_once instead of a mutex

Reply:
How about the following?
```
tatic std::map<const char *, int> *ZeCallCount = [] {
  if (ZeDebug & ZE_DEBUG_CALL_COUNT) {
    return new std::map<const char *, int>;
  }
  return (std::map<const char *, int>*)nullptr;
}();
```

This global static variable will be initialized only once when pi_level_zero is loaded.

Reply:
That's what I initially suggested, but it is possibly that pi_level_zero will not be used after loaded (and this it's piTearDown will not be called to delete the map). I think it is more robust to move this lambda to piPlatformsGet and call it under std::call_once

Reply:
Done. Thanks.